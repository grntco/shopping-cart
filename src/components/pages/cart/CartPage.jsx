import { Link } from 'react-router-dom'

const CartPage = () => {
    return (
        <>
            <h1>Cart Page!</h1>
            <Link to='/'>Go home?</Link>
        </>
    )
}

export default CartPage
