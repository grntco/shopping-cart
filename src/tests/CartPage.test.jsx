import { describe, it, expect } from 'vitest'
import { render, screen, waitFor } from '@testing-library/react'
import { BrowserRouter } from 'react-router-dom'
import userEvent from '@testing-library/user-event'
import CartPage from '../components/pages/cart/CartPage'

const mockCart = [
    {
        id: 1,
        title: 'Fjallraven - Foldsack No. 1 Backpack, Fits 15 Laptops',
        price: 109.95,
        description:
            'Your perfect pack for everyday use and walks in the forest. Stash your laptop (up to 15 inches) in the padded sleeve, your everyday',
        category: "men's clothing",
        image: 'https://fakestoreapi.com/img/81fPKd-2AYL._AC_SL1500_.jpg',
        rating: {
            rate: 3.9,
            count: 120,
        },
    },
]

describe('Cart Page test', () => {
    it('renders correctly with an empty cart', () => {
        render(<CartPage cart={[]} handleDeleteFromCart={() => {}} />, {
            wrapper: BrowserRouter,
        })
        expect(screen.getByTestId('empty-cart-page')).toMatchSnapshot()
        expect(
            screen.getByRole('heading', {
                level: 2,
                name: 'Your Cart is Empty',
            }),
        ).toBeInTheDocument()
    })

    it('renders correctly with a full cart', () => {
        render(<CartPage cart={mockCart} handleDeleteFromCart={() => {}} />, {
            wrapper: BrowserRouter,
        })
        expect(screen.getByTestId('cart-page')).toMatchSnapshot()
        expect(
            screen.getByRole('heading', {
                level: 2,
                name: 'Your Cart',
            }),
        ).toBeInTheDocument()
        expect(
            screen.getByRole('heading', {
                level: 2,
                name: 'Summary',
            }),
        ).toBeInTheDocument()
    })
})
