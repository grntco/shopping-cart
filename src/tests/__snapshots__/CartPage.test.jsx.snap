// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Cart Page test > renders correctly with a full cart 1`] = `
<section
  class="sc-gLLuof eenBgs"
  data-testid="cart-page"
>
  <div
    class="sc-iBdnpw glbRni"
  >
    <h2>
      Your Cart
    </h2>
    <ul
      class="sc-fsYfdN jLwFgZ"
    >
      <li
        class="sc-dmyCSP boPveX"
      >
        <a
          href="/products/1"
        >
          <img
            alt="Fjallraven - Foldsack No. 1 Backpack, Fits 15 Laptops"
            src="https://fakestoreapi.com/img/81fPKd-2AYL._AC_SL1500_.jpg"
          />
        </a>
        <div
          class="sc-hLQSwg ftDPRI"
        >
          <div>
            <a
              href="/products/1"
            >
              <h4>
                Fjallraven - Foldsack No. 1 Backpack, Fits 15 Laptops
              </h4>
            </a>
            <p>
              $109.95
            </p>
          </div>
          <div
            class="sc-eDLKkx fChlpz"
          >
            <p>
              Qty: 
              1
            </p>
            <button
              class="sc-guDLey KDEhA"
            >
              <img
                alt="trash icon"
                src="/src/assets/icons/trash-2.svg"
                title="Delete all of this item"
              />
            </button>
          </div>
        </div>
      </li>
    </ul>
  </div>
  <div
    class="sc-iBdnpw glbRni"
  >
    <h2>
      Summary
    </h2>
    <p>
      Woohoo! You qualify for free shipping!
    </p>
    <div
      class="sc-qZrbh dFjnbG"
    >
      <div>
        <p>
          Subtotal
        </p>
        <p>
          $109.95
        </p>
      </div>
      <div>
        <p>
          Free Shipping
        </p>
        <p>
          $0.00
        </p>
      </div>
      <div>
        <p>
          Total
        </p>
        <p>
          $109.95
        </p>
      </div>
    </div>
    <button
      class="sc-beySPh gepDOA"
    >
      <img
        alt="lock icon"
        color="white"
        src="/src/assets/icons/lock.svg"
      />
      Checkout Securely
    </button>
  </div>
</section>
`;

exports[`Cart Page test > renders correctly with an empty cart 1`] = `
<section
  class="sc-jTQCzO izRpYK"
  data-testid="empty-cart-page"
>
  <h2>
    Your Cart is Empty
  </h2>
  Add products to your cart to see them appear here!
</section>
`;
